packages/core/src/Stores/base-store.js

```diff
@@ -285,56 +285,62 @@ export default class BaseStore {
 
     @action.bound
     onSwitchAccount(listener) {
-        this.switchAccountDisposer = when(
-            () => this.root_store.client.switch_broadcast,
-            async () => {
-                try {
-                    const result = this.switch_account_listener();
-                    if (result && result.then && typeof result.then === 'function') {
-                        result.then(() => {
-                            this.root_store.client.switchEndSignal();
-                            this.onSwitchAccount(this.switch_account_listener);
-                        });
-                    } else {
-                        throw new Error('Switching account listeners are required to return a promise.');
-                    }
-                } catch (error) {
-                    // there is no listener currently active. so we can just ignore the error raised from treating
-                    // a null object as a function. Although, in development mode, we throw a console error.
-                    if (!isProduction()) {
-                        console.error(error); // eslint-disable-line
+        if (listener) {
+            this.switch_account_listener = listener;
```

Comment:
do we need to assign the listener here?

could we instead:
```javascript
const result = listener()
...
this.onSwitchAccount(listener);
```

Reply:
No, we don't need to call listener early.

Reply:
@salar-fs sorry I don't understand what you mean, can you explain again?